client-misk/src/main/kotlin/app/cash/backfila/client/misk/embedded/Backfila.kt

```diff
@@ -9,29 +10,45 @@ import okio.ByteString
  * the Backfila dashboard UI.
  */
 interface Backfila {
-  fun <Type : Backfill<*, *>> createDryRun(
+  fun <Type : Backfill<*, *, *>> createDryRun(
     backfill: KClass<Type>,
     parameters: Map<String, ByteString>,
     rangeStart: String?,
     rangeEnd: String?
   ): BackfillRun<Type>
 
-  fun <Type : Backfill<*, *>> createWetRun(
+  fun <Type : Backfill<*, *, *>> createWetRun(
     backfillType: KClass<Type>,
     parameters: Map<String, ByteString>,
     rangeStart: String?,
     rangeEnd: String?
   ): BackfillRun<Type>
 }
 
-inline fun <reified Type : Backfill<*, *>> Backfila.createDryRun(
-  parameters: Map<String, ByteString> = mapOf(),
+inline fun <reified Type : Backfill<*, *, *>> Backfila.createDryRun(
+  parameters: Any? = null,
+  parameterData: Map<String, ByteString> = mapOf(),
   rangeStart: String? = null,
   rangeEnd: String? = null
-): BackfillRun<Type> = createDryRun(Type::class, parameters, rangeStart, rangeEnd)
+): BackfillRun<Type> {
+  check(parameterData.isEmpty() || parameters == null) {
+    "Only one of parameters and parameterData can be set"
```

Comment:
plan to remove the parameterData later?

Reply:
No, I want to give you the option to provide it as a raw bytestring. You can only specify one though. Hence the check